src/backends/caffe/caffeinputconns.h

```diff
@@ -342,11 +342,15 @@ namespace dd
 		return;
 	      }
 	      // create db
-	      images_to_db(_uris,_model_repo + "/" + _dbname,_model_repo + "/" + _test_dbname);
+	      if (!this->_unchanged_data)
+		images_to_db(_uris,_model_repo + "/" + _dbname,_model_repo + "/" + _test_dbname);
+	      else images_to_db(_uris,_model_repo + "/" + _dbname,_model_repo + "/" + _test_dbname,
+				"lmdb",false,"");
 	      
 	      // compute mean of images, not forcely used, depends on net, see has_mean_file
-	      compute_images_mean(_model_repo + "/" + _dbname,
-				  _model_repo + "/" + _meanfname);
+	      if (!this->_unchanged_data)
```

Comment:
from the above comment, I am not sure the test is enough... maybe a `todo` ?

Reply:
Good catch on the `computer_mean`, it should be active for images. Not sure we want to remove the mean on spectrograms. 

The test is not thorough but should be enough to avoid the mean computation when data should not be unchanged I guess.

Reply:
so in the future, should it be active on many-channel images?

Reply:
Yes I've removed the `compute_mean` since it was using uint8 operations and I should reactivate it before merge!

Reply:
so, the test should be on `float` type rather than on the `unchanged` flag, am I right?

Reply:
We need a single flag that says don't touch the data, it's not a true image inside, i.e. don't encode it, etc... Whether it's float or not actually. We already have something similar inside caffe when we're storing labels into images (e.g. for segmentation).